app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -394,8 +412,315 @@ public void onTaskFailure(String result) {
+                        streamActv.setError(getString(R.string.stream_not_exists));
+                        streamActv.requestFocus();
+                        return;
+                    }
+                } catch (SQLException e) {
+                    Log.e("SQLException", "SQL not correct", e);
+                }
+            }
+            if (TextUtils.isEmpty(topicActv.getText().toString())) {
+                topicActv.setError(getString(R.string.subject_error));
+                topicActv.requestFocus();
+                return;
+            }
+        } else {
+            if (TextUtils.isEmpty(streamActv.getText().toString())) {
+                streamActv.setError(getString(R.string.person_error));
+                streamActv.requestFocus();
+                return;
+            }
+        }
+
+        if (TextUtils.isEmpty(messageEt.getText().toString())) {
+            messageEt.setError(getString(R.string.no_message_error));
+            messageEt.requestFocus();
+            return;
+        }
+        sendingMessage(true);
+        MessageType messageType = (isCurrentModeStream()) ? MessageType.STREAM_MESSAGE : MessageType.PRIVATE_MESSAGE;
+        Message msg = new Message(app);
+        msg.setSender(app.getYou());
+
+        if (messageType == MessageType.STREAM_MESSAGE) {
+            msg.setType(messageType);
+            msg.setStream(new Stream(streamActv.getText().toString()));
+            msg.setSubject(topicActv.getText().toString());
+        } else if (messageType == MessageType.PRIVATE_MESSAGE) {
+            msg.setType(messageType);
+            msg.setRecipient(topicActv.getText().toString().split(","));
+        }
+        msg.setContent(messageEt.getText().toString());
+        AsyncSend sender = new AsyncSend(that, msg);
+        sender.setCallback(new ZulipAsyncPushTask.AsyncTaskCompleteListener() {
+            public void onTaskComplete(String result, JSONObject jsonObject) {
+                Toast.makeText(ZulipActivity.this, R.string.message_sent, Toast.LENGTH_SHORT).show();
+                messageEt.setText("");
+                sendingMessage(false);
+            }
+            public void onTaskFailure(String result) {
+                Log.d("onTaskFailure", "Result: " + result);
+                Toast.makeText(ZulipActivity.this, R.string.message_error, Toast.LENGTH_SHORT).show();
+                sendingMessage(false);
+            }
+        });
+        sender.execute();
     }
 
+    private void sendingMessage(Boolean isSending) {
+        streamActv.setEnabled(!isSending);
+        textView.setEnabled(!isSending);
+        messageEt.setEnabled(!isSending);
+        topicActv.setEnabled(!isSending);
+        sendBtn.setEnabled(!isSending);
+        togglePrivateStreamBtn.setEnabled(!isSending);
+        if (isSending)
+            composeStatus.setVisibility(View.VISIBLE);
+        else
+            composeStatus.setVisibility(View.GONE);
+    }
+
+    SimpleCursorAdapter streamActvAdapter;
```

Comment:
best practice is to put these declarations at the top of the file with all other member variables.